mhkit/river/io/d3d.py

```diff
@@ -1,127 +1,310 @@
-from os.path import abspath, dirname, join, isfile, normpath, relpath
 from mhkit.utils import unorm
-from math import isclose
 import scipy.interpolate as interp
-import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
 import netCDF4
+import warnings
 
 
-def get_layer_data(data, variable, layer_index= -1 , time_index=-1):
+def get_all_time(data):
     '''
-    Get variable data from netcdf4 object at specified layer and timestep. 
+    Returns all of the data time stamps in seconds.
+    
+    Parameters
+    ----------
+    data: NetCDF4 object 
+       A NetCDF4 object that contains spatial data, e.g. velocity or shear
+       stress generated by running a Delft3D model.  
+
+    Returns
+    -------
+    seconds_run: array
+        Returns an array of integers representing the number of seconds after
+        the simulation started and that the data object contains a snapshot of
+        simulation conditions at that time.
+    '''
+    
+    assert type(data)== netCDF4._netCDF4.Dataset, 'data must be NetCDF4 object' 
+
+    seconds_run = np.ma.getdata(data.variables['time'][:], False)
+
+    return seconds_run
+
+def index_to_seconds(data, time_index):
+    '''
+    The function will return 'seconds_run' if passed a 'time_index' 
+
+    Parameters
+    ----------
+    data: NetCDF4 object 
+       A NetCDF4 object that contains spatial data, e.g. velocity or shear
+       stress, generated by running a Delft3D model.  
+    time_index: int 
+        A positive integer to pull the time index from the dataset. 0 being closest
+        to time 0. Default is last time index -1.
+
+    Returns
+    -------
+    seconds_run: int, float
+        The 'seconds_run' is the seconds corresponding to the 'time_index' increments.
+    '''
+    return _convert_time(data, time_index=time_index)
+
+
+def seconds_to_index(data, seconds_run):
+    '''
+    The function will return a 'time_index' if passed a 'seconds_run'
```

Comment:
The function will return the nearest 'time_index' in the data if passed an integer number of 'seconds_run'